src/mjolnir/ingest_transit.cc

```diff
@@ -394,20 +406,30 @@ float get_stop_pair_dist(const gtfs::Stop& stop_connect,
 }
 
 // return dangling stop_pairs, write stop data from feed
-bool write_stop_pair(Transit& tile,
-                     const tile_transit_info_t& tile_info,
-                     const feed_object_t& feed_trip,
-                     const gtfs::Feed& feed,
-                     const std::unordered_map<feed_object_t, GraphId>& platform_node_ids,
-                     unique_transit_t& uniques) {
+bool write_stop_pair(
+    Transit& tile,
+    const tile_transit_info_t& tile_info,
+    const feed_object_t& feed_trip,
+    const gtfs::Feed& feed,
+    const std::unordered_map<feed_object_t, GraphId>& platform_node_ids,
+    unique_transit_t& uniques,
+    const google::protobuf::RepeatedPtrField<valhalla::mjolnir::Transit_Node>& tile_nodes) {
   bool dangles = false;
 
   const auto& tile_tripId = feed_trip.id;
   const std::string currFeedPath = feed_trip.feed;
 
   const auto& currTrip = feed.get_trip(tile_tripId);
-  const auto& currShape = feed.get_shape(currTrip->shape_id);
   const auto& trip_calendar = feed.get_calendar(currTrip->service_id);
+  if (!currTrip || !trip_calendar) {
+    LOG_ERROR("Feed " + feed_trip.feed + ", trip ID" + tile_tripId +
+              " can't be found or has no calendar.txt entry, skipping...");
+    return false;
+  }
+
+  // get the gtfs shape and our pbf shape_id if present
+  const auto& currShape = feed.get_shape(currTrip->shape_id);
+  auto pbf_shape_it = tile_info.shapes.find({currTrip->shape_id, feed_trip.feed});
```

Comment:
so we can get its tile-specific shape_id

Reply:
I'm very proud of myself BTW that I didn't rename `currShape` at this point hahaha